### DISCUSSION TITLE
SqlClient.dll Assembly with SqlServer and Dbatools.library module

### DISCUSSION THREAD

**Discussion by harshthkkr** [dataplat/dbatools#8916] [Q&A] [answered]
We have the following modules installed on our system:

Name Version
SqlServer 22.0.59
dbatools 2.0.0
dbatools.library 2023.5.5

After the new release was installed when we imported the module, it gave the below error
import-module dbatools
Exception: Couldn't import C:\Program Files\PowerShell\Modules\dbatools.library\2023.5.5\core\lib\win-sqlclient\Microsoft.Data.SqlClient.dll | Assembly with same name is already loaded

I found the DLL is present on the path: (Module-SqlServer)
PowerShell\Modules\SqlServer\22.0.59\coreclr\runtimes\win\lib\netcoreapp3.1\Microsoft.Data.SqlClient.dll

Does anyone know the workaround for this?

**Comment by potatoqualitee** [comment, marked as answer]
We dont support this setup for reasons like this :/ but some people have luck importing dbatools first and sqlserver second.

**Reply by harshthkkr** [reply]
Thanks @potatoqualitee. I did the same. It worked, so I closed the discussion
